### PR TITLE
Improved Timer 

### PR DESCRIPTION
**Pull request by gaurav274**
Add a context manager based timer class

### CODE REVIEW

File: eva/utils/timer.py [hunk 1]
```diff
@@ -17,38 +17,29 @@
...
+    def __enter__(self):
+        assert self._start_time is None, "Concurrent calls are not supported"
+        self._start_time = time.perf_counter()
 
-def end_timer():
-    global _START_TIME
-    global _END_TIME
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        assert self._start_time is not None, "exit called with starting the context"
+        time_elaped = time.perf_counter() - self._start_time
```
Comment (jarulraj) [inline eva/utils/timer.py:36]: Minor typo: elapsed

File: eva/utils/timer.py [hunk 2]
```diff
@@ -17,38 +17,29 @@
...
-_START_TIME = None
-_END_TIME = None
 
+class Timer:
+    """Class used for logging time metrics.
 
-def start_timer():
-    global _START_TIME
-    global _END_TIME
+    This is not thread safe"""
```
Comment (jarulraj) [inline eva/utils/timer.py:24]: What are the implications of this -- like when not to use this? We could maybe add that as a comment.

**Comment by jarulraj** [comment]
This is much better 👍 Does not corrupt the CI log with timing log records.

File: eva/utils/timer.py (hunk 2 above)
Comment (gaurav274) [reply eva/utils/timer.py:24]: There isn't an example in the system right now. I added a comment so that in the future when we bring ray, we are careful.

**Pull request merged**